boot/bootutil/src/boot_record.c

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2018-2020 Arm Limited
+ * Copyright (c) 2020 Linaro Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "mcuboot_config/mcuboot_config.h"
+
+#ifdef MCUBOOT_SHARE_BOOT_RECORD
+#include "bootutil/boot_record.h"
+#include "bootutil/boot_status.h"
+#include "bootutil_priv.h"
+#include "bootutil/image.h"
+#include "flash_map_backend/flash_map_backend.h"
+
+/**
+ * @enum shared_data_err_t
+ *
+ * @brief Return values for adding data entry to shared memory area
+ */
+enum shared_memory_err_t {
```

Comment:
This is a fairly different convention from the rest of MCUBoot. The value is local to just this file, but it does end up resulting in needing an 'rc2' later. The convention is still that '0' is success, and I wonder if we'd be better just using int, and defines for the errors themselves.

Reply:
Yes, you are right.
I've eliminated this new `enum` type from the PR and now it uses only defines and `int` as a return type.